loader/loader.go

```diff
@@ -1,38 +1,100 @@
 package loader
 
 import (
+	"bytes"
 	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"strings"
 
 	"github.com/docker/app/internal"
+	"github.com/docker/app/specification"
 	"github.com/docker/app/types"
+	"github.com/docker/cli/cli/compose/loader"
+	"github.com/docker/cli/cli/compose/schema"
 	"github.com/docker/docker/pkg/archive"
 	"github.com/pkg/errors"
 )
 
+var (
+	crlf       = []byte{'\r', '\n'}
+	lf         = []byte{'\n'}
+	delimiters = [][]byte{
+		[]byte("\r\n---\r\n"),
+		[]byte("\n---\r\n"),
+		[]byte("\r\n---\n"),
+		[]byte("\n---\n"),
+	}
+)
+
+// useCRLF detects which line break should be used
+func useCRLF(data []byte) bool {
+	nbCrlf := bytes.Count(data, crlf)
+	nbLf := bytes.Count(data, lf)
+	switch {
+	case nbCrlf == nbLf:
+		// document contains only CRLF
+		return true
+	case nbCrlf == 0:
+		// document does not contain any CRLF
+		return false
+	default:
+		// document contains mixed line breaks, so let use the OS preferency
```

Comment:
`// document contains mixed line breaks, so let use the OS preferency`

`// document contains mixed line breaks so use the OS default`